File: tests/rosetta/transpiler/Pascal/call-a-function-12.pas
{$mode objfpc}
program Main;
uses SysUtils;
type FuncType1 = function(b: integer): integer;
type FuncType2 = function(p0: integer): integer;
type FuncType3 = function(y: integer): integer;
var
  anon0_cap0: integer;
  anon4_cap0: integer;
  main_add2: FuncType2;
  main_add3: FuncType2;
  main_partial: FuncType2;
function anon0(b: integer): integer;
begin
  exit(anon0_cap0 + b);
end;
function anon0_make(a: integer): FuncType1;
begin
  anon0_cap0 := a;
  exit(anon0);
end;
function mkAdd(a: integer): FuncType2;
begin
  exit(anon0_make(a));
end;
function mysum(x: integer; y: integer): integer;
begin
  exit(x + y);
end;
function anon4(y: integer): integer;
begin
  exit(mysum(anon4_cap0, y));
end;
function anon4_make(x: integer): FuncType3;
begin
  anon4_cap0 := x;
  exit(anon4);
end;
function partialSum(x: integer): FuncType2;
begin
  exit(anon4_make(x));
end;
procedure main();
begin
  main_add2 := mkAdd(2);
  main_add3 := mkAdd(3);
  writeln((IntToStr(add2(5)) + ' ') + IntToStr(add3(6)));
  main_partial := partialSum(13);
  writeln(IntToStr(partial(5)));
end;
begin
  main();
end.

File: tests/rosetta/transpiler/Clojure/call-a-function-12.clj
(ns main (:refer-clojure :exclude [mkAdd mysum partialSum main]))

(require 'clojure.set)

(defn in [x coll]
  (cond (string? coll) (clojure.string/includes? coll x) (map? coll) (contains? coll x) (sequential? coll) (some (fn [e] (= e x)) coll) :else false))

(defn padStart [s w p]
  (loop [out (str s)] (if (< (count out) w) (recur (str p out)) out)))

(def nowSeed (atom (let [s (System/getenv "MOCHI_NOW_SEED")] (if (and s (not (= s ""))) (Integer/parseInt s) 0))))

(declare mkAdd mysum partialSum main)

(declare main_add2 main_add3 main_partial)

(defn mkAdd [mkAdd_a]
  (try (throw (ex-info "return" {:v (fn [b] (+ mkAdd_a mkAdd_b))})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn mysum [mysum_x mysum_y]
  (try (throw (ex-info "return" {:v (+ mysum_x mysum_y)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn partialSum [partialSum_x]
  (try (throw (ex-info "return" {:v (fn [y] (mysum partialSum_x partialSum_y))})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn main []
  (do (def main_add2 (mkAdd 2)) (def main_add3 (mkAdd 3)) (println (str (str (str (add2 5)) " ") (str (add3 6)))) (def main_partial (partialSum 13)) (println (str (partial 5)))))

(defn -main []
  (main))

(-main)
